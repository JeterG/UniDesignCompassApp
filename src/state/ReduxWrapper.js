import React from 'react';
import { Provider } from 'react-redux';
import createStore from "./store";
import { PersistGate } from 'redux-persist/integration/react'

export default ({ element }) => (
  <Provider store={createStore().store}>
    <PersistGate loading={null} persistor={createStore().persistor}>
      {element}
    </PersistGate>
  </Provider>
);