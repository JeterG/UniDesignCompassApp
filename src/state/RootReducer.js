import { combineReducers } from 'redux';
import Reducer from './reducers';

export default combineReducers({ state: Reducer });
