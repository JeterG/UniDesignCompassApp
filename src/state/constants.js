export const AUTH_USER = "AUTH_USER";
export const UPDATE_USER = "UPDATE_USER";
export const LOGOUT_USER = "LOGOUT_USER";
