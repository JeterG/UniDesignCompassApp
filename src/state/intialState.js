export const initialState = {
  user : {},
  isAuthenticated: false,
}