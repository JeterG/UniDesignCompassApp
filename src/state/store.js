import {  createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import RootReducer from './RootReducer';

const composeEnhancers = (typeof window !== `undefined`) ? (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose ) : compose;

export default function configureStore() {
  return createStore(
    RootReducer,
    composeEnhancers(applyMiddleware(thunk))
  );
 }