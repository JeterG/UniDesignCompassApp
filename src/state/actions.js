import { FETCH_USER_INFO, AUTH_USER } from "./constants"

export const fetchUserInfo = (orderID) => ({
  type: FETCH_USER_INFO,
  payload: orderID
})

export const authenticateUser = (authenticated = false) => dispatch => {

  dispatch(
    {
      type: AUTH_USER,
      payload: "true",
      aws_user: [1]
    }
  )
}
