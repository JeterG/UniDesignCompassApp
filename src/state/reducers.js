import {initialState} from "./intialState"
import {AUTH_USER} from "./constants"

export default (state = initialState, action) => {
  switch(action.type) {
    case AUTH_USER :
      return {
        ...state,
        user: action.payload,
        isAuthenticated: true
      };
    default:
      return state;
  }
}

