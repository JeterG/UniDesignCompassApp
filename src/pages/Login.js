import React, { Component } from "react";
import Layout from "../components/layout"
import "../components/bootstrap.css"
import { connect } from 'react-redux';
import { authenticateUser } from '../state/actions';
import { Auth } from "aws-amplify";
import config from "../aws-exports";
import NewPassword from "../components/newPassword";
import SignIn from "../components/SignIn";
import SignUp from "../components/SignUp";
Auth.configure(config);

class Login extends Component {
    constructor(props) {
        super(props);
        this.state = {
            email: "",
            password: "",
            repeat_pass:"",
            phone:"",
            username:"",
            user: [],
        };
        this.handleAuth = this.handleAuth.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this._comp = null;
        this.Log_state = "SignIn"
    }

    handleChange = (e)=>{
        this.setState({[e.target.id]:e.target.value});
    }
    handleAuth = (e) => {
        e.preventDefault();
        Auth.signIn(this.state.email, this.state.password)
            .then((res) => {
                switch (res.challengeName) {
                    case "NEW_PASSWORD_REQUIRED":
                        this.Log_state = "NEWPASS";
                        this.setState({
                            user:res
                        })
                        break;
                    case "SOFTWARE_TOKEN_MFA":
                        this.Log_state = "OTP";
                        break;
                    case "MFA_SETUP":
                        break;
                    case "SMS_MFA":
                        break;
                    default:
                        console.log(res);
                        this.Log_state = "LoggedIn";
                        this.props.authenticateUser("true");
                }
                this.forceUpdate();

            },(error)=>{
                switch(error.code){
                    case "UserNotFoundException":
                        alert("User not found Make an account");
                        break;
                    default:

                }
            });
    };
    handlePass=(e)=>{
        e.preventDefault();
        console.log(this.state.user)
        Auth.completeNewPassword(this.state.user,this.state.password)
        .then((res)=>{
            console.log(res);
        },(error)=>{
            console.log(error)
        })
    }
    createAccount=(e)=>{
        e.preventDefault();
        this.Log_state="SIGNUP";
        this.forceUpdate();
    }
    handleCreate=(e)=>{
        e.preventDefault();
        let attributes={username:this.state.username,password:this.state.password,attributes:{email:this.state.email,phone_number:this.state.phone}};
        console.log(attributes);
        Auth.signUp(attributes)
        .then((res)=>{
            this.Log_state="Verify";
        },(error)=>{
            console.log(error);
        })
    }
    determineRender() {
        console.log(this.Log_state);
        switch (this.Log_state) {
            case 'SignIn':
                this._comp = <SignIn handleAuth={this.handleAuth} handleChange={this.handleChange} handleAccount={this.createAccount}/>
                break;
            case 'NEWPASS':
                this._comp=<NewPassword handlePass={this.handlePass} handleChange={this.handleChange}/>
                break;
            case 'MFA_SETUP':
                break;
            case 'OTP':
                break;
            case 'SIGNUP':
                this._comp =<SignUp handleChange={this.handleChange} handleCreate={this.handleCreate} />
                break;
            default:
        }
    }
        render() {
            this.determineRender()
            return (<Layout>{
                this._comp
            }</Layout>
            )
        }
    }
    
function mapStateToProps(state) {
    return {
        isAuthenticated: state.Reducer.isAuthenticated,
    }
}
const mapDispatchToProps = dispatch => ({
    authenticateUser: (auth) => dispatch(authenticateUser(auth))
})
export default connect(mapStateToProps, mapDispatchToProps)(Login);
