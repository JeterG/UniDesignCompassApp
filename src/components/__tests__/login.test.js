import React from "react"
import renderer from "react-test-renderer"
import Login from "../../pages/Login";
import { store } from "../../state/store";
import { Provider } from 'react-redux';
import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {mount} from 'enzyme';
import {mountToJson} from 'enzyme-to-json';
configure({adapter:new Adapter()});

const tree = mount(<Provider store={store} ><Login /></Provider>);

describe("Check Login_Page", () => {
    it("render", () => {
        expect(mountToJson(tree)).toMatchSnapshot()
    })
})
describe("Check Password", () => {
    it("render", () => {
        expect(mountToJson(tree)).toMatchSnapshot()
    })
})
describe("Check verification page loads", () => {
    it("render", () => {
        expect(mountToJson(tree)).toMatchSnapshot()
    })
})
describe("Check user signup page ", () => {
    it("render", () => {
        expect(mountToJson(tree)).toMatchSnapshot()
    })
})
describe("", () => {
    it("render", () => {
        expect(mountToJson(tree)).toMatchSnapshot()
    })
})

// describe("SignIn exceptions",() =>{
//      it("failed Log in",()=>{
//          console.log();
//         tree.find('[type="submit"]');
//      })
// })
