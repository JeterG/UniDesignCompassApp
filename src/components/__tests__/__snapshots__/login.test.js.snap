// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Login_Page render 1`] = `<Connect(Login) />`;

exports[`Check Password render 1`] = `<Connect(Login) />`;

exports[`Check user signup page  render 1`] = `<Connect(Login) />`;

exports[`Check verification page loads render 1`] = `<Connect(Login) />`;

exports[`render 1`] = `<Connect(Login) />`;
